### PR TITLE
Office improvements

### PR DESCRIPTION
**Pull request by icebreaker**
@brandonc @mhuggins

### CODE REVIEW

File: lib/wopinator/sanitizer.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      def sanitize(s)
+        s.gsub(/[^a-zA-Z0-9\.\-\_\s]/, '_')
+      end
+
+      # RFC 2152 (UTF-7)
+      def decode(s)
+        s.gsub(/\+([A-Za-z0-9\/]*)-?/) { decode_chunk(Regexp.last_match[1]) }
+      end
+
+      def decode_sanitize(s)
```
Comment (mhuggins) [inline lib/wopinator/sanitizer.rb]: ~~I don't see this being used anywhere.~~  Never mind, I see it's being used by wintessa.

File: lib/wopinator/discovery.rb [hunk 2]
```diff
@@ -104,6 +107,24 @@ def parse_url(url)
...
+      {}.tap do |hash|
+        params.each do |k, v|
+          if key = supported[k]
+            hash[key] = v
+          end
+        end
+      end
+    end
+
+    def extract_supported_action_query_params(url)
```
Comment (brandonc) [inline lib/wopinator/discovery.rb]: 🗒 Would love some comments explaining what this is doing

Comment (icebreaker) [inline lib/wopinator/discovery.rb]: This is then used to add back to the URL things like `IsLicensedUser=1` by replacing BUSINESS_USER with `1`.

Comment (icebreaker) [inline lib/wopinator/discovery.rb]: @brandonc Added this to the code as well: https://github.com/kapost/wopinator/commit/84e2fe52cbc09e203e8f3f7711a6fd5a0db389fb

**Pull request merged**